src/widgets/RulerPanel.h

```diff
@@ -0,0 +1,82 @@
+﻿/**********************************************************************
+
+  Audacity: A Digital Audio Editor
+
+  RulerPanel.h
+
+  Dominic Mazzoni
+
+**********************************************************************/
+
+#ifndef __AUDACITY_RULER_PANEL__
+#define __AUDACITY_RULER_PANEL__
+
+#include "Ruler.h"
```

Comment:
The fourth commit splits RulerPanel from Ruler.

Observe that including `Ruler.h` in `RulerPanel.h` is unavoidable, because there is a nested `Ruler` object.

Ok then -- but that means the transitive inclusion of `Ruler.h` happens whever `RulerPanel.h` is included.

There are four such inclusions, including that in `RulerPanel.cpp`.  I suggest then you simplify the lists of `#inlude` directives in each place.